src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,90 @@
+package duke.task;
+
+import java.time.LocalDate;
+
+/**
+ * Represents a task that the user wants to do. Can set to completed and uncompleted.
+ *
+ */
+public abstract class Task {
+    protected String objective;
+    protected boolean done;
+
+    /**
+     * Constructor for Task class where Task is by default uncompleted.
+     * @param objective Task to be done.
+     */
+    public Task(String objective) {
+        this.objective = objective;
+        this.done = false;
+    }
+
+    /**
+     * Constructor for Task class where Task is by default uncompleted.
+     * @param objective Task to be done.
+     * @param done Whether Task is completed.
+     */
+    public Task(String objective, Boolean done) {
+        this.objective = objective;
+        this.done = done;
+    }
+
+    /**
+     * Task is completed. Mark it with a cross.
+     */
+    public void mark() {
+        this.done = true;
+    }
+
+    /**
+     * Task is uncompleted. It is unmarked.
+     */
+    public void unmarked() {
+        this.done = false;
+    }
+
+    /**
+     * Returns a serialized version of the string.
+     *
+     * @return Serialized form of the string.
+     */
+    public abstract String serialize();
+
+
+    public boolean contains(String keyword) {
```

Comment:
Maybe can name this method as 'hasKeyword'. Though I think 'contains' is intuitive enough.